test/extended/networking/apply_commatrix_firewall.go

```diff
@@ -0,0 +1,136 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift-kni/commatrix/commatrix"
+	"github.com/openshift-kni/commatrix/consts"
+	"github.com/openshift-kni/commatrix/debug"
+
+	clientutil "github.com/openshift-kni/commatrix/client"
+	nodesutil "github.com/openshift-kni/commatrix/nodes"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	g.It("should apply firewall by open ports on nodes", func() {
+		artifactsDir := filepath.Join(exutil.ArtifactDirPath(), "commatrix")
+		err := os.MkdirAll(artifactsDir, 0755)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("generating the commatrix")
+		kubeconfig, ok := os.LookupEnv("KUBECONFIG")
+		if !ok {
+			g.Fail("must set the KUBECONFIG environment variable")
+		}
+		// Get the current working directory
+		dir, err := os.Getwd()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		fmt.Println(" getting current working directory:", dir)
+
+		cs, err := clientutil.New(kubeconfig)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		deployment := commatrix.MNO
+		isSNO, err := isSNOCluster(cs.ConfigV1Interface)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deployment = commatrix.SNO
+		}
+		format := "nft"
+		mat, err := commatrix.New(kubeconfig, "", "", commatrix.Cloud, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		err = commatrix.WriteMatrixToFileByType(*mat, "communication-matrix", format, deployment, dir)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		ns := consts.DefaultDebugNamespace + "firewall"
+		err = debug.CreateNamespace(cs, ns)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		defer func() {
+			err := debug.DeleteNamespace(cs, ns)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}()
+		g.By("apply firewall on master nodes")
+
+		localFilePath := dir + "/communication-matrix-master.nft"
+		fmt.Println(" local path:", localFilePath)
+		fileContent, err := os.ReadFile(localFilePath)
```

Comment:
Can you generate the nftables format and use it instead of writing to file and reading?